src/meta/src/hummock/compaction/picker/space_reclaim_compaction_picker.rs

```diff
@@ -91,96 +62,99 @@ impl SpaceReclaimCompactionPicker {
         state: &mut SpaceReclaimPickerState,
     ) -> Option<CompactionInput> {
         assert!(!levels.levels.is_empty());
-        let reclaimed_level = levels.levels.last().unwrap();
         let mut select_input_ssts = vec![];
-        let level_handler = &level_handlers[reclaimed_level.level_idx as usize];
-
-        if reclaimed_level.table_infos.is_empty() {
-            // no file to be picked
-            state.clear();
-            return None;
-        }
-
-        if state.valid()
-            && state
-                .last_select_end_bound
-                .compare_right_with(&state.end_bound_in_round.right)
-                == std::cmp::Ordering::Greater
-        {
-            // in round but end_key overflow
-            // turn to next_round
-            state.clear();
-            return None;
+        if let Some(l0) = levels.l0.as_ref() && state.last_level == 0 {
+            // only pick trivial reclaim sstables because this kind of task could be optimized and do not need send to compactor.
+            for level in &l0.sub_levels {
+                for sst in &level.table_infos {
+                    let exist_count = self.exist_table_count(sst);
+                    if exist_count == sst.table_ids.len() ||  level_handlers[0].is_pending_compact( &sst.sst_id) {
+                        if !select_input_ssts.is_empty() {
+                            break;
+                        }
+                    } else if exist_count == 0 {
+                        select_input_ssts.push(sst.clone());
+                    } else if !select_input_ssts.is_empty() {
+                        break;
+                    }
+                }
+                if !select_input_ssts.is_empty() {
+                    return Some(CompactionInput {
+                        input_levels: vec![
+                            InputLevel {
+                                level_idx: level.level_idx,
+                                level_type: level.level_type,
+                                table_infos: select_input_ssts,
+                            },
+                            InputLevel {
+                                level_idx: 0,
+                                level_type: level.level_type,
+                                table_infos: vec![],
+                            },
+                        ],
+                        target_level: level.level_idx as usize,
```

Comment:
is level.level_idx always 0?

Reply:
Sure, this branch is only handle the `levels.l0`